test/kubemark/start-kubemark.sh

```diff
@@ -47,21 +47,115 @@ gcloud compute instances create ${MASTER_NAME} \
+
+
+echo "bearer token = ${KUBE_BEARER_TOKEN}"
+echo "${CA_CERT}" | base64 -d > ca.crt
+echo "${KUBECFG_CERT}" | base64 -d > kubecfg.crt
+echo "${KUBECFG_KEY}" | base64 -d > kubecfg.key
 
 until gcloud compute ssh hollow-cluster-master --command="ls" &> /dev/null; do
   sleep 1
 done
 
+gcloud compute ssh hollow-cluster-master --command="sudo mkdir /srv/kubernetes -p && \
+sudo bash -c \"echo ${MASTER_CERT} | base64 -d > /srv/kubernetes/server.cert\" && \
+sudo bash -c \"echo ${MASTER_KEY} | base64 -d > /srv/kubernetes/server.key\" && \
+sudo bash -c \"echo ${CA_CERT} | base64 -d > /srv/kubernetes/ca.crt\" && \
+sudo bash -c \"echo ${KUBECFG_CERT} | base64 -d > /srv/kubernetes/kubecfg.crt\" && \
+sudo bash -c \"echo ${KUBECFG_KEY} | base64 -d > /srv/kubernetes/kubecfg.key\" && \
+sudo bash -c \"echo \"${KUBE_BEARER_TOKEN},admin,admin\" > /srv/kubernetes/known_tokens.csv\" && \
+sudo bash -c \"echo \"${KUBELET_TOKEN},kubelet,kubelet\" >> /srv/kubernetes/known_tokens.csv\" && \
+sudo bash -c \"echo \"${KUBE_PROXY_TOKEN},kube_proxy,kube_proxy\" >> /srv/kubernetes/known_tokens.csv\" && \
+sudo bash -c \"echo admin,admin,admin > /srv/kubernetes/basic_auth.csv\""
+
 gcloud compute copy-files \
-  ${KUBE_ROOT}/_output/release-tars/kubernetes-server-linux-amd64.tar.gz \
-  ${KUBE_ROOT}/test/kubemark/start-kubemark-master.sh \
-  ${KUBE_ROOT}/test/kubemark/configure-kubectl.sh \
-  hollow-cluster-master:~
+    ${KUBE_ROOT}/_output/release-tars/kubernetes-server-linux-amd64.tar.gz \
+    ${KUBE_ROOT}/test/kubemark/start-kubemark-master.sh \
+    ${KUBE_ROOT}/test/kubemark/configure-kubectl.sh \
+    hollow-cluster-master:~
 
-gcloud compute ssh hollow-cluster-master --command="chmod a+x configure-kubectl.sh && chmod a+x start-kubemark-master.sh && ./start-kubemark-master.sh"
+gcloud compute ssh hollow-cluster-master --command="chmod a+x configure-kubectl.sh && chmod a+x start-kubemark-master.sh && sudo ./start-kubemark-master.sh"
+
+# create kubeconfig for Kubelet:
+KUBECONFIG_CONTENTS=`echo "apiVersion: v1
+kind: Config
+users:
+- name: kubelet
+  user:
+    client-certificate-data: ${KUBELET_CERT}
+    client-key-data: ${KUBELET_KEY}
+clusters:
+- name: kubemark
+  cluster:
+    certificate-authority-data: ${CA_CERT}
+    server: https://${MASTER_IP}
+contexts:
+- context:
+    cluster: kubemark
+    user: kubelet
+  name: kubemark-context
+current-context: kubemark-context" | base64 | tr -d "\n\r"`
+
+KUBECONFIG_SECRET=kubeconfig_secret.json
+cat > "${KUBECONFIG_SECRET}" << EOF
+{
+  "apiVersion": "v1",
+  "kind": "Secret",
+  "metadata": {
+    "name": "kubeconfig"
+  },
+  "type": "Opaque",
+  "data": {
+    "kubeconfig": "${KUBECONFIG_CONTENTS}"
+  }
+}
+EOF
+
+LOCAL_KUBECONFIG=${KUBE_ROOT}/test/kubemark/kubeconfig.loc
+cat > "${LOCAL_KUBECONFIG}" << EOF
+apiVersion: v1
+kind: Config
+users:
+- name: admin
+  user:
+    client-certificate-data: ${KUBECFG_CERT}
+    client-key-data: ${KUBECFG_KEY}
+    username: admin
+    password: admin
```

Comment:
It'd be much better to have the password something autogenerated and random (like the service account tokens) rather than a fixed string.